.github/workflows/test.yml

```diff
@@ -20,9 +20,9 @@ jobs:
       - uses: actions/cache@v3
         with:
           path: ~/.conan2
-          key: ${{runner.os}}-conan-${{ hashFiles('**/conanfile.py') }}
+          key: ubuntu-latest-conan-${{ hashFiles('**/conanfile.py') }}
```

Comment:
Is there a way to make this test workflow just a subset of the tasks in the publish workflow? In other words, just do some of the matrix jobs, and not publish?

Reply:
We probably could, but that seems like wasted effort to me. This file makes sure that any code merged to the repo compiles and passes lint checks. Why bother making sure it does that on multiple CPU architectures, especially if we throw away the compiled binary afterwards?

Reply:
Because there are three different compilers with three different standard libraries:
- MSVC for Windows, with the associated standard library
- AppleClang for macOS, with libc++
- G++ on Linux, with libstdc++

It is very easy to write code that works on one of these but breaks one of the others, and without testing them at PR time you won't find out you broke things until release time, when it is too late.

Reply:
I think this is a mistake because it will add a 12-minute delay every time you push a commit. but I'll do it because you seem to think it's that important.